src/visualization/components/internal/LegendOptions.tsx

```diff
@@ -137,9 +138,17 @@ export const OpacitySlider: FC<OpacitySliderProps> = ({
   handleSetOpacity,
   testID = 'opacity-slider',
 }) => {
-  // without the toFixed(0) sometimes you
-  // can get numbers like 45.000009% which we want to avoid
-  const percentLegendOpacity = (legendOpacity * 100).toFixed(0)
+  let validOpacity = legendOpacity
+  if (
+    typeof validOpacity !== 'number' ||
+    validOpacity !== validOpacity ||
+    validOpacity < LEGEND_OPACITY_MINIMUM ||
+    validOpacity > LEGEND_OPACITY_MAXIMUM
```

Comment:
what do you think about inverting this? i think inverting the meaning will show it being a default a little better.

```ts
let validOpacity = LEGEND_OPACITY_DEFAULT
  if (
    typeof legendOpacity === 'number' &&
    legendOpacity === legendOpacity &&
    legendOpacity > LEGEND_OPACITY_MINIMUM &&
    legendOpacity < LEGEND_OPACITY_MAXIMUM
  ) {
    validOpacity = legendOpacity
  }

Reply:
Sure thing. This reads better.